### DISCUSSION TITLE
JIT is breaking themes build with Gatsby Themes and Yarn Workspaces

### DISCUSSION THREAD

**Discussion by DennisKraaijeveld** [tailwindlabs/tailwindcss#7669] [Help]
**What version of Tailwind CSS are you using?**
v2.2.19

**What build tool (or framework if it abstracts the build tool) are you using?**

Gatsby V4.1

**What version of Node.js are you using?**

V16.12.0

**What browser are you using?**

Brave & Chrome

**What operating system are you using?**

Mac OS Monterrey Mac Mini M1

**Reproduction URL**

https://github.com/DennisKraaijeveld/tailwind-jit-reproduction

It is a Yarn Workspace to develop Gatsby Themes. 

**Describe your issue**

I am moving over my base project to a Gatsby Theme. On my base project, everything is working like expected but right now I am running into issues. The issue is when you enable 'jit' mode on the website where the theme is installed, that it will break the theme and all css will disappear.  First I thought it was an issue between CSS and SCSS, but this didnt matter and I found that 'jit' was actually the thing which was breaking everything. 

When disabling 'jit' on the site dir, everything works, but arbitrary styles from the theme dir, are not working anymore, since this needs JIT to be enabled. I do have JIT enabled in the tailwind-config in my theme dir, but this config file seems to be there just for fun.

**STEPS**

1. Run `yarn workspaces example develop`
2. Enable mode: 'jit' on `tailwind-config.js` inside the example dir to see what happens

**Comment by DennisKraaijeveld** [comment]
I was thinking I could rewrite some of the stuff to get it to work for now by disabling JIT, but also all the `after` and `before` classes are obviously not working.. 

Got it working for now by letting the site watch the files in the theme directory. But is this expected behaviour? Maybe site A will not use some of the templates of the theme, will CSS then still be created during build?

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

When I follow the commands you provided (after struggling with yarn/berry for a little bit), the site looks like this to me: 
<img width="1031" alt="image" src="https://user-images.githubusercontent.com/1834413/140555307-166b35e8-6bf3-4cf8-84dc-67ca5d782e69.png">

I do see styles, but I'm not sure if that is what you meant with broken or not. If this is the expected behaviour, make sure to remove any `gatsby` caches and restart your server.

That said, one thing I noticed is that the `tailwind.config.js` file contains `'./src/**/*.{js,jsx,ts,tsx}'` in the `purge` config. But there is no `src` folder in the `example` folder so there is nothing to generate. It could be that gatsby is doing something magical behind the scenes (because I do have _some_ output).

In general, make sure that your paths you list actually exist on disk. Another thing you can play with is to make sure to look in all folders. You could play with this for example:
```js
module.exports = {
  purge: [process.env.PROJECT_CWD + './*/src/**/*.{js,jsx,ts,tsx}'],
  // ...
}
```

But I would start with cleaning gatsby caches, since I definitely see **something**.

**Comment by DennisKraaijeveld** [comment]
If you enable `jit` in the example folder, the styles will disappear. But I do understand now why this is happening.

Gatsby Themes let you build a theme which you can install as a NPM package. Small quote from their website:


> Gatsby themes are plugins that include a gatsby-config.js file and add pre-configured functionality, data sourcing, and/or UI code to Gatsby sites. You can think of Gatsby themes as separate Gatsby sites that can be put together and allow you to split up a larger Gatsby project!


My Theme has all the templates and pages and I was thinking that during build or develop, these templates will get created inside the src folder of the website. Apparently this is not the case and I have to set `purge` to watch the src folder of my theme.

Do I have to take care of something else since all the styling is inside a NPM package and is a dependency of the website? Or will everything be fine

**Comment by adamwathan** [comment]
Going to convert this to a discussion since it seems to be more of a configuration/usage question.

You definitely need to make sure your `content` configuration includes every file in your project that contains Tailwind classes, even if those files are in a dependency and living in `node_modules`:

```js
// tailwind.config.js
module.exports = {
  content: [
    './node_modules/my-theme/src/**/*.js',
  ],
}
```

...or similar 👍
